src/System.Threading.Thread/src/System/Threading/Thread.cs

```diff
@@ -2,28 +2,122 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#if !NETNATIVE
```

Comment:
I'm assuming the NETNATIVE define will go away completely by the next release. If some differences are expected to remain, I believe it is preferred in CoreFX to make it a partial class and use different files for runtime or OS-related differences, instead of using preprocessor defines.